sentry-android-core/src/main/java/io/sentry/android/core/util/RootChecker.java

```diff
@@ -0,0 +1,147 @@
+package io.sentry.android.core.util;
+
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import io.sentry.core.ILogger;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.util.Objects;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class RootChecker {
+
+  /** the UTF-8 Charset */
+  @SuppressWarnings("CharsetObjectCanBeUsed")
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  private RootChecker() {}
+
+  /**
+   * Check if the device is rooted or not
+   * https://medium.com/@thehimanshugoel/10-best-security-practices-in-android-applications-that-every-developer-must-know-99c8cd07c0bb
```

Comment:
`h`: The whole class is basically the same as in the article here. I would prefer if we have tests for this. I think we would have to change the code a little, but still, I would prefer it. If we don't it is a Blackbox that we just have to trust and it is hard to change, because you don't know if you break anything.

Reply:
@philipphofmann ideas on how to test if a device is rooted or not? this is just heuristics and tests won't be deterministic if we run tests on the JVM and not on a real rooted/non-rooted device.

Reply:
I can't mock a `Runtime.getRuntime()` for example

Reply:
I can test a few parts of this PR though, not the whole thing.
I'm blocked by #420
I'd need `IBuildInfoProvider` to be able to mock eg `Build.TAGS`

Reply:
I think we would need to split the class up in a few components. We could create a class named `BuildTagsProvider`. I wouldn't put this functionality into `IBuildInfoProvider` because BuildTags is more of an Android thing and also this class most likely is only going to stay on Android.

We could delegate `Runtime.getRuntime().exec(su)` to another class. Then we could return a process of our choice and we could make sure that the `BufferedReader` and the error handling is working fine.

Reply:
For the `PackageManager` we could create our own function or class that checks whether a package exists and write actual integration tests to make sure if it is working fine.

Reply:
@philipphofmann for PackageManager only mocking is fine, no need to create extra classes.

Reply:
I did test all the cases, but I'm generally against creating classes for everything just for the sake of readability/testability.
I've created a private package ctor that takes all the necessary params, so I can mock and test them isolated.